### PR TITLE
TLSN MVP

### PR DESCRIPTION
**Pull request by sinui0**
This PR implements the minimal functionality required to notarize a session. It does not implement anything for selective disclosure.

### CODE REVIEW

File: tlsn/tlsn-notary/src/config.rs [hunk 1]
```diff
@@ -0,0 +1,30 @@
+const DEFAULT_MAX_TRANSCRIPT_SIZE: usize = 2 << 14; // 16Kb
```
Comment (themighty1) [inline tlsn/tlsn-notary/src/config.rs:1]: 1 << 14

File: tlsn/tlsn-prover/src/config.rs [hunk 2]
```diff
@@ -0,0 +1,96 @@
+use actor_ot::{OTActorReceiverConfig, OTActorSenderConfig};
+use mpc_share_conversion::{ReceiverConfig, SenderConfig};
+use tls_client::RootCertStore;
+use tls_mpc::{MpcTlsCommonConfig, MpcTlsLeaderConfig};
+
+const DEFAULT_MAX_TRANSCRIPT_SIZE: usize = 2 << 14; // 16Kb
```
Comment (themighty1) [inline tlsn/tlsn-prover/src/config.rs:6]: 1 << 14

**Review by themighty1 — APPROVED** [review]
gw!

**Review by th4s — APPROVED** [review]
nice :rocket:, ack!

File: tlsn/tlsn-notary/src/lib.rs [hunk 3]
```diff
@@ -0,0 +1,257 @@
...
+impl Future for NotaryBackgroundFut {
+    type Output = Result<(), NotaryError>;
+
+    fn poll(
+        mut self: std::pin::Pin<&mut Self>,
+        cx: &mut std::task::Context<'_>,
+    ) -> std::task::Poll<Self::Output> {
+        self.fut.poll_unpin(cx)
+    }
+}
```
[the remark this answers is not in the document]
Comment (th4s) [reply tlsn/tlsn-notary/src/lib.rs:52]: Ahh ok, what benefit is that?

**Pull request merged**